### DISCUSSION TITLE
How to customize the OAuth Login and Swagger UI to accept custom fields.

### DISCUSSION THREAD

**Discussion by basiltt** [fastapi/fastapi#8617] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class OAuth2ClientCredentialsRequestForm:
    def __init__(
            self,
            grant_type: str = Form(None, regex="client_credentials"),
            scope: str = Form(""),
            client_id: Optional[str] = Form(None),
            client_secret: Optional[str] = Form(None),
    ):
        self.grant_type = grant_type
        self.scopes = scope.split()
        self.client_id = client_id
        self.client_secret = client_secret
```

### Description

I do have a special use case where I want to build a custom OAuth authentication with 3 fields client_id, client_secret and username.  I would like to know how to customize swagger UI to accept these 3 fields along with scopes.

### Operating System

Windows

### Operating System Details

Windows 10 20H2

### FastAPI Version

0.75.0

### Python Version

Python 3.9.9

**Comment by basiltt** [comment]
![custom_login_modified](https://user-images.githubusercontent.com/54134686/157191703-a3bcd5dc-72e1-4a93-87eb-6c180edd4bbc.png)

It should be something like this

**Comment by skysantoroa** [comment]
Did you find any solution?
Also `client_id` returns None always when requesting form data of type `OAuth2PasswordRequestForm`

**Comment by diegolramirez** [comment]
Hey! I am also looking for a workaround, having this exact situation. Have you found any solutions? Thanks

**Comment by yashp280793** [comment]
Hello, I am also looking for this solution.

**Comment by basiltt** [comment]
I couldn't find a solution for it yet. I'm currently managing it with client credential location as body. and passing client id and client secret.

**Comment by YuriiMotov** [comment]
Similar to https://github.com/fastapi/fastapi/discussions/4793
